### ISSUE TITLE
Using "os" as theme does not update when OS theme changes

### ISSUE THREAD

**Issue by tcobbs-bentley** [iTwin/iTwinUI-react#485]
#### Describe the bug (actual behavior)
If you use ThemeProvider with them set to "os" or useTheme("os"), it correctly detects the color theme at the time of component creation. However, it does not detect on-the-fly changes in the OS color theme.

#### Expected behavior
If a user uses "os" as the theme, ThemeProvider and useTheme() should both detect when the OS theme changes and update accordingly.

#### Reproduction
https://codesandbox.io/s/itwinui-react-example-forked-7fvtc?file=/src/App.tsx

##### Steps to reproduce
Load the sandbox. Note that its theme matches the OS theme.
Change your OS theme.
Notice that the theme does not change in the sandbox.

Note: I tested the above sandbox on macOS Big Sur. My actual code is running on iOS, where the same problem occurs.

#### Additional information
I have code in another location that detects changes to the OS color them. The code looks something like this:

```ts
    try {
      window.matchMedia("(prefers-color-scheme: dark)").addEventListener("change", MobileUi._colorSchemeListener);
    } catch (e) {
      // Safari didn't support the above BASIC functionality until version 14.
      // eslint-disable-next-line deprecation/deprecation
      window.matchMedia("(prefers-color-scheme: dark)").addListener(MobileUi._colorSchemeListener);
    }
```

The source file that the above code came from can be found here:

https://github.com/iTwin/mobile-ui-react/blob/main/src/mobile-ui-react/MobileUi.tsx

**Comment by tcobbs-bentley** [comment]
I forgot to mention when talking about the addEventListener, but I (perhaps obviously) have other code to remove the listener when done. That would go into the return value of whatever useEffect is used to call addEventListener, and has the same try/catch block, but with removeEventListener in the try and removeListener in the catch.

**Comment by tcobbs-bentley** [comment]
@mayank99 Your solution is going to fail on any iOS device prior to iOS 14, since they implement addListener/removeListener _instead of_ addEventListener/removeEventListener. I realize that your ?. checks on the functions will prevent it from throwing an exception at runtime, but since you did that instead of the try/catch I used in my code, it will silently fail there. That means that I have to continue to provide my workaround, since I unfortunately have to support iOS 13 (probably until iOS 16 is released).

**Comment by mayank99** [comment]
Yes, I realize it will "fail" on older Safari versions. That is a conscious decision made with graceful progressive enhancement in mind. It's a relatively minor thing as the user can just refresh the page to get the correct theme. But you can choose to add your own fallback if you really need this edge case.

**Comment by tcobbs-bentley** [comment]
I didn't mention in the report, but my bug report was for the mobile use case where imodeljs is used inside a native app (in a built-in web view). In this use case, they almost certainly cannot "just refresh", since they hopefully aren't even aware that what they see is running in a browser view. So my workaround of only using dark and light (and doing the media query myself) will have to stand. I'm not saying you have to fix it, just that your fix isn't usable to me.
